test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -436,4 +445,15 @@ protected void ensureNoInitializingShards() {
         assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());
     }
 
+    /**
+     * Iff this returns true concurrent search is favour by making more probable to generate multiple concurrent slices.
+     * Otherwise runs with default set up. The default is randomized.
+     */
+    protected boolean eagerConcurrentSearch() {
+        if (eagerConcurrentSearch == null) {
+            eagerConcurrentSearch = randomBoolean();
+        }
+        return eagerConcurrentSearch;
```

Comment:
I wonder if it'd make sense to start aggressively testing concurrency, and having true as a default rather than random. And then tests that are problematic can disable it? Otherwise we could rarely run sequentially, but most times use concurrency.

Reply:
Also, I am not sure how overriding this in subclasses would work. The variable is private and there is no way for subclasses to set it? Adding the ability to override the default behaviour could also be a thing that we add once we need it.

Reply:
These comments are valid also for ESIntegTestCase.